### PR TITLE
SMILE-1181-bug-calculating-page-count-for-journals

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

File: src/Components/Dialogs/ConfirmImportDialog.js [hunk 1]
```diff
@@ -5,6 +5,25 @@ import TextField from '@material-ui/core/TextField';
...
+    pageFrom === null ||
+    pageTo === null ||
+    !isNaN(pageTo) ||
+    !isNaN(pageFrom) ||
+    !reg.test(pageTo.toString()) ||
+    !reg.test(pageFrom.toString())
+  ) {
+    return '0';
+    //Javascript håndterer 2^53 bit mønster, SQL hånterer 2^31 for integers.
+  } else if (pageTo - pageFrom > Math.pow(2, 31)) {
```
Comment (pcb-unit) [inline src/Components/Dialogs/ConfirmImportDialog.js:20]: trenger vel egentlig ikke denne med den regexpen ?

File: src/Components/Dialogs/ConfirmImportDialog.js [hunk 2]
```diff
@@ -5,6 +5,25 @@ import TextField from '@material-ui/core/TextField';
 import { patchPiaPublication, patchPublication, postPublication } from '../../api/publicationApi';
 import { handlePotentialExpiredSession } from '../../api/api';
 
+const getNumberOfPages = (pageFrom, pageTo) => {
+  const reg = new RegExp('^[0-9]+$');
+  if (
+    pageFrom === null ||
+    pageTo === null ||
+    !isNaN(pageTo) ||
+    !isNaN(pageFrom) ||
```
Comment (pcb-unit) [inline src/Components/Dialogs/ConfirmImportDialog.js:14]: trenger kanskje ikke denne heller med regexpen.

File: src/Components/Dialogs/ConfirmImportDialog.js [hunk 3]
```diff
@@ -5,6 +5,25 @@ import TextField from '@material-ui/core/TextField';
 import { patchPiaPublication, patchPublication, postPublication } from '../../api/publicationApi';
 import { handlePotentialExpiredSession } from '../../api/api';
 
+const getNumberOfPages = (pageFrom, pageTo) => {
+  const reg = new RegExp('^[0-9]+$');
```
Comment (pcb-unit) [inline src/Components/Dialogs/ConfirmImportDialog.js:9]: kan legge inn en kommentar på  hvorfor vi gjør det slik - det glemmer man fort

File: src/Components/Dialogs/ConfirmImportDialog.js [hunk 4]
```diff
@@ -5,6 +5,25 @@ import TextField from '@material-ui/core/TextField';
...
+  if (
+    pageFrom === null ||
+    pageTo === null ||
+    !isNaN(pageTo) ||
+    !isNaN(pageFrom) ||
+    !reg.test(pageTo.toString()) ||
+    !reg.test(pageFrom.toString())
```
Comment (pcb-unit) [inline src/Components/Dialogs/ConfirmImportDialog.js:16]: hva med versjonen : `numberMatcher=/^[0-9]+$/i;  pageFrom.match(numberMatcher)  `

File: src/Components/Dialogs/ConfirmImportDialog.js [hunk 5]
```diff
@@ -232,6 +245,7 @@ export default function ConfirmImportDialog(props) {
       }}
       disableEscapeKeyDown>
       <DialogTitle>Bekreft import</DialogTitle>
+      <pre style={{ maxWidth: '90%' }}>{JSON.stringify(createPublicationObject(), null, 2)}</pre>
```
Comment (pcb-unit) [inline src/Components/Dialogs/ConfirmImportDialog.js:248]: du glemte denne :-)

File: src/Components/Dialogs/ConfirmImportDialog.js [hunk 6]
```diff
@@ -122,13 +141,7 @@ export default function ConfirmImportDialog(props) {
...
         from: publication.channel.pageFrom,
         to: publication.channel.pageTo,
-        count:
-          publication.channel.pageTo !== null &&
-          publication.channel.pageFrom !== null &&
-          !isNaN(publication.channel.pageTo) &&
-          !isNaN(publication.channel.pageFrom)
-            ? (publication.channel.pageTo - publication.channel.pageFrom).toString()
-            : '0',
+        count: getNumberOfPages(publication.channel.pageFrom, publication.channel.pageTo),
```
Comment (pcb-unit) [inline src/Components/Dialogs/ConfirmImportDialog.js:144]: nice

**Review by pcb-unit — CHANGES_REQUESTED** [review]

File: src/Components/Dialogs/ConfirmImportDialog.js (hunk 1 above)
Comment (anetteOlli) [reply src/Components/Dialogs/ConfirmImportDialog.js:20]: PageTo: 345345352345345345345345341232323 PageFrom 0. vil gå igjennom regex'en og isNAN, men krasje backenden.

File: src/Components/Dialogs/ConfirmImportDialog.js (hunk 5 above)
Comment (anetteOlli) [reply src/Components/Dialogs/ConfirmImportDialog.js]: takk, takk

File: src/Components/Dialogs/ConfirmImportDialog.js (hunk 4 above)
Comment (anetteOlli) [reply src/Components/Dialogs/ConfirmImportDialog.js]: fikset

File: src/Components/Dialogs/ConfirmImportDialog.js (hunk 2 above)
Comment (anetteOlli) [reply src/Components/Dialogs/ConfirmImportDialog.js]: fikset

**Review by pcb-unit — DISMISSED** [review]

File: src/Components/Dialogs/ConfirmImportDialog.js [hunk 7]
```diff
@@ -8,8 +8,8 @@ import { handlePotentialExpiredSession } from '../../api/api';
 const getNumberOfPages = (pageFrom, pageTo) => {
   //Dette har blitt lagt inn pga. edge-casen med sidetall med bokstaven "e" i seg.
   //pageTo 2086.e8, pageFrom: 2083
-  const reg = new RegExp('^[0-9]+$');
-  if (pageFrom === null || pageTo === null || !reg.test(pageTo.toString()) || !reg.test(pageFrom.toString())) {
+  const reg = /^[0-9]+$/i;
```
Comment (pcb-unit) [inline src/Components/Dialogs/ConfirmImportDialog.js:11]: hva med arabicNumberMatcher ?
